app/navigation/DocumentsNavigator.scala

```diff
@@ -85,26 +89,34 @@ class DocumentsNavigator @Inject()(implicit frontendAppConfig: FrontendAppConfig
     case _ => routes.SessionExpiredController.onPageLoad()
   }
 
-
   private def whenGoverningDocumentApprovedPageNav(userAnswers: UserAnswers, mode: Mode): Call = userAnswers.get(WhenGoverningDocumentApprovedPage) match {
     case Some(_) if mode == CheckMode => regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
     case Some(_) => regulatorDocsRoutes.IsApprovedGoverningDocumentController.onPageLoad(mode)
     case _ => routes.SessionExpiredController.onPageLoad()
   }
 
   private def isApprovedGoverningDocumentPageNav(userAnswers: UserAnswers, mode: Mode): Call = userAnswers.get(IsApprovedGoverningDocumentPage) match {
+    case Some(true) if mode == CheckMode => userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage) match {
+      case Some(_) if userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage).isDefined =>
+        regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
+      case _ => regulatorDocsRoutes.HasCharityChangedPartsOfGoverningDocumentController.onPageLoad(mode)
+    }
```

Comment:
do we need this pattern match here? wouldn't 
`      if (userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage).isDefined) {
        regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
      } else {
        regulatorDocsRoutes.HasCharityChangedPartsOfGoverningDocumentController.onPageLoad(mode)
      }
` do the same thing?

Reply:
fixed